### PR TITLE
Bugfix for improper check related to virtual crates

### PR DESCRIPTION
**Pull request by mosteo**

### CODE REVIEW

File: src/alire/alire-solutions.adb [hunk 1]
```diff
@@ -1202,7 +1202,7 @@ package body Alire.Solutions is
       end if;
 
       for Dep of This.Dependencies loop
-         if Dep.Has_Release and then Dep.Release.Provides (Dep.Crate) then
+         if Dep.Has_Release and then Dep.Release.Provides (Crate) then
```
Comment (mosteo) [inline src/alire/alire-solutions.adb:1205]: This was the actual bug, ouch.

**Review by Fabien-Chouteau — APPROVED** [review]

**Comment by Fabien-Chouteau** [comment]
I forgot in the review, but is there a tests that we can add to make sure doesn't happen again?

**Comment by mosteo** [comment]
I'll prepare a reproducer, but given the new assertions in the code the root problem will have to be from other quarters for this to happen again.

**Comment by Fabien-Chouteau** [comment]
@mosteo I feel like I hit this bug a lot. Should we make a patch release for it?

**Comment by mosteo** [comment]
Indeed. Only let me check that #880 is not affecting the 1.1 branch because that one would also nice to have fixed.

**Pull request merged**